src/gentropy/l2g.py

```diff
@@ -160,26 +165,92 @@ def __init__(
         self.feature_matrix = L2GFeatureMatrix(
             _df=session.load_data(feature_matrix_path), features_list=self.features_list
         )
-        self.variant_index = (
-            VariantIndex.from_parquet(session, variant_index_path)
-            if variant_index_path
-            else None
-        )
 
         if run_mode == "predict":
             self.run_predict()
         elif run_mode == "train":
-            self.gs_curation = (
-                self.session.spark.read.json(gold_standard_curation_path)
-                if gold_standard_curation_path
-                else None
-            )
-            self.interactions = (
-                self.session.spark.read.parquet(gene_interactions_path)
-                if gene_interactions_path
-                else None
-            )
-            self.run_train()
+            gold_standard = self.prepare_gold_standard()
+            self.run_train(gold_standard)
+
+    def prepare_gold_standard(self) -> L2GGoldStandard:
```

Comment:
I like the logic in general. We could also have 2 different params: one for the raw GS, and another one for the complete GS, but it could be confusing.

The only problem I have is expecting that the complete GS comes in JSON format to match the old curation format. We internally work with Parquets everywhere. If anything, I'll convert the old file to Parquet